File: cpp/1991.find-the-middle-index-in-array.cpp
class Solution {
public:
    int solve(vector<int> &nums){
	int n = nums.size();
	int _sum=0;
	vector<int> _right(n, 0);
	for(int i = n-1; i>=0; i--){
	       	_right[i] = _sum;
	       	_sum+=nums[i];
	}
	_sum = 0;
	for(int i = 0 ; i < n; i++){
	   if( _right[i] == _sum ) return  i;
	   _sum += nums[i];
	}
	return -1;
    }
    int findMiddleIndex(vector<int>& nums) {
	return solve(nums);
    }
};

File: python/1991.find-the-middle-index-in-array.py
class Solution:
    def findMiddleIndex(self, nums: List[int]) -> int:
        n = len(nums)
        _right = list(reversed(list(accumulate(reversed(nums[1:] + [0])))))
        for i in range(n):
            if _right[i] == _sum:
                return i
            _sum += nums[i]
        return -1
